frontend/javascripts/oxalis/view/settings/setting_input_views.js

```diff
@@ -297,48 +286,92 @@ export class Vector6InputSetting extends React.PureComponent<
     // only numbers, commas and whitespace is allowed
     const isValidInput = /^[\d\s,]*$/g.test(text);
     const value = Utils.stringToNumberArray(text);
-    const isValidFormat = value.length === 6 || value.length === 0;
-
-    if (isValidFormat && isValidInput) {
-      if (value.length === 0) {
-        this.props.onChange();
-      } else {
-        this.props.onChange(Utils.numberArrayToVector6(value));
-      }
+    const isValidLength = value.length === 6;
+    const isValid = isValidInput && isValidLength;
+
+    if (isValid) {
+      this.props.onChange(Utils.numberArrayToVector6(value));
     }
+    this.setState({ text, isValid });
+  };
 
-    this.setState({
-      text,
-      isValid: isValidInput && isValidFormat,
-    });
+  handleNameChange = (evt: SyntheticInputEvent<>) => {
+    this.props.onChange(undefined, evt.target.value);
```

Comment:
You could make the argument to `onChange` an object, and only supply the key and value you actually want to set - to get rid of these undefineds here.